include/ecs/system_manager.h

```diff
@@ -0,0 +1,49 @@
+#ifndef ECS_SYSTEM_MANAGER_H
+#define ECS_SYSTEM_MANAGER_H
+
+#include <vector>
+#include "common.h"
+#include "system.h"
+
+namespace arch::ecs {
+	class SystemManager {
+	public:
+		template <typename T>
+		std::shared_ptr<T> add_system();
+
+		template <typename T>
+		void set_system_component_mask(ComponentMask mask);
+
+		void on_entity_destroyed(EntityId entity_id);
+		void on_entity_mask_change(EntityId entity_id, ComponentMask entity_mask);
+
+	private:
+		std::vector<std::shared_ptr<System>> _systems;

```

Comment:
Using shared_ptr there means that SystemManager doesn't actually own systems. Is it something we actually want here? If system is destroyed in SystemManager, but some other object has it's shared_ptr, it still stays alive. Can we in our use case get away with just returning system references?

Reply:
I don't really know about pointers, so I would need a direct solution XD